jobs/transforms/skill_validation_test.py

```diff
@@ -0,0 +1,289 @@
+                skill_validation.ValidateSkillCommitCmdsSchema())
+        )
+
+        self.assert_pcoll_equal(output, [
+            base_validation_errors.CommitCmdsValidateError(
+                invalid_commit_cmd_model,
+                {'cmd': 'invalid'},
+                'Command invalid is not allowed')
+        ])
+
+    def test_skill_change_object_with_missing_attribute_in_cmd(self):
+        invalid_commit_cmd_model = skill_models.SkillCommitLogEntryModel(
+            id='123',
+            created_on=self.YEAR_AGO,
+            last_updated=self.NOW,
+            commit_type='test-type',
+            user_id='',
+            skill_id='123',
+            post_commit_status='private',
+            commit_cmds=[{
+                'cmd': 'update_skill_property',
+                'property_name': 'name',
+            }])
+
+        output = (
+            self.pipeline
+            | beam.Create([invalid_commit_cmd_model])
+            | beam.ParDo(
+                skill_validation.ValidateSkillCommitCmdsSchema())
+        )
+
+        self.assert_pcoll_equal(output, [
+            base_validation_errors.CommitCmdsValidateError(
+                invalid_commit_cmd_model,
+                {
+                    'cmd': 'update_skill_property',
+                    'property_name': 'name',
+                },
+                'The following required attributes are missing: '
+                'new_value, old_value')
+        ])
+
+    def test_skill_change_object_with_extra_attribute_in_cmd(self):
+        invalid_commit_cmd_model = skill_models.SkillCommitLogEntryModel(
+            id='123',
+            created_on=self.YEAR_AGO,
+            last_updated=self.NOW,
+            commit_type='test-type',
+            user_id='',
+            skill_id='123',
+            post_commit_status='private',
+            commit_cmds=[{
+                'cmd': 'add_skill_misconception',
+                'new_misconception_dict': {
+                    'id': 0,
+                    'name': 'name',
+                    'notes': '<p>notes</p>',
+                    'feedback': '<p>default_feedback</p>'
+                },
+                'invalid': 'invalid'
+            }])
+
+        output = (
+            self.pipeline
+            | beam.Create([invalid_commit_cmd_model])
+            | beam.ParDo(
+                skill_validation.ValidateSkillCommitCmdsSchema())
+        )
+
+        # Why can't I use dict here.
+        self.assert_pcoll_equal(output, [
+            base_validation_errors.CommitCmdsValidateError(
+                invalid_commit_cmd_model,
+                '{u\'cmd\': u\'add_skill_misconception\', '
+                'u\'new_misconception_dict\': {u\'id\': 0, u\'notes\': '
+                'u\'<p>notes</p>\', u\'feedback\': '
+                'u\'<p>default_feedback</p>\', u\'name\': u\'name\'}, '
+                'u\'invalid\': u\'invalid\'}',
+                'The following extra attributes are present: invalid')
+        ])
```

Comment:
I have made the same comments in two other places too.
If I don't pass in a string then It doesn't pass test. Is it ok to leave it be for now ?

Reply:
Can you please attach the log, as to what it is showing when this is a dict.

Reply:
here is the log
![image](https://user-images.githubusercontent.com/29734619/119529423-b8a1e700-bd9f-11eb-9084-fc6dd92eed75.png)

its the same expected dict but  new_misconception_dict inside is not in order.

Reply:
In that case, just reorder it accordingly and check if the test passes.

Reply:
It doesn't work. I have tried reordering different ways, i cannot match the expected output.

Reply:
Would it be possible to only keep the `new_misconception_dict` value as str and the rest as dict?

Reply:
@vojtechjelinek done PTAL!